### PR TITLE
use os.execl to "run" subcommand on Unix ( close #315 )

### LINKED ISSUE #315: Add `exec` subcommand to use PIPE and REDIRECTION ?
PIPE and REDIRECTION are not working with `pipenv run`, because this uses ~~`pexpect.spawn`~~ 
 `pexpect.spawn.wait` with non-interactive environment.

```shell
$ pipenv run python3 -c "print('foo')" | grep foo
# => not found
$ pipenv run python3 -c "print('foo')" > bar.txt
# => bar.txt is empty
```

If use `os.exec` instead of `spawn`, pipe and redirection are available.

```diff
diff --git a/pipenv/cli.py b/pipenv/cli.py
index 62a4165..f48b9e7 100644
--- a/pipenv/cli.py
+++ b/pipenv/cli.py
@@ -997,7 +997,8 @@ def run(command, args, no_interactive=False, three=None, python=False):

     # Spawn the new process, and interact with it.
     try:
-        c = pexpect.spawn(which(command), list(args))
+        os.execv(which(command), [command] + list(args))
+        # c = pexpect.spawn(which(command), list(args))
     except pexpect.exceptions.ExceptionPexpect:
         click.echo(crayons.red('The command ({0}) was not found within the virtualenv!'.format(which(command))))
         sys.exit(1)
```

```console
$ python3 -m pipenv run python3 -c "print('foo')" | grep foo
foo
$ python3 -m pipenv run python3 -c "print('foo')" > bar.txt
$ cat bar.txt
foo
```

Can you add `exec` subcommand like bundler, or change `run` command to use `os.exec` ?

## version

3.5.6, or current master (7b17058)

### PR DESCRIPTION
**Pull request by nonylene**
see #315.

Pipe and redirection are working well in my Linux and Windows computer.

### CODE REVIEW

File: pipenv/cli.py [hunk 1]
```diff
@@ -984,38 +984,26 @@ def sigwinch_passthrough(sig, data):
...
         p.communicate()
         sys.exit(p.returncode)
-
-    # Interact with the new shell.
-    if no_interactive:
-        c.wait()
     else:
-        c.interact()
-        c.close()
-    sys.exit(c.exitstatus)
```
Comment (nateprewitt) [inline pipenv/cli.py]: When `os.execl` runs and the process fails, will this return an appropriate exit status?

File: pipenv/cli.py [hunk 2]
```diff
@@ -984,38 +984,26 @@ def sigwinch_passthrough(sig, data):
...
-    # Spawn the new process, and interact with it.
-    try:
-        c = pexpect.spawn(which(command), list(args))
-    except pexpect.exceptions.ExceptionPexpect:
-        click.echo(crayons.red('The command ({0}) was not found within the virtualenv!'.format(which(command))))
+    if not os.path.exists(command_path):
+        click.echo(crayons.red('The command ({0}) was not found within the virtualenv!'.format(command_path)))
         sys.exit(1)
 
-    # Windows!
```
Comment (nateprewitt) [inline pipenv/cli.py]: This is a minor nit, but any reason to move this comment inside the conditional? This project tends to have comments precede the code they're describing.

File: pipenv/cli.py (hunk 1 above)
Comment (nonylene) [inline pipenv/cli.py]: yes.

```sh
$ python3 -m pipenv run python -c "exit()"

$ python3 -m pipenv run python -c "exit(1)"
zsh: exit 1     python3 -m pipenv run python -c "exit(1)"

$ python3 -m pipenv run python -c "exit(2)"
zsh: exit 2     python3 -m pipenv run python -c "exit(2)"

$ python3 -m pipenv run python -c "exit(3)"
zsh: exit 3     python3 -m pipenv run python -c "exit(3)"

$ python3 -m pipenv run python -c "exit(3"
  File "<string>", line 1
    exit(3
         ^
SyntaxError: unexpected EOF while parsing
zsh: exit 1     python3 -m pipenv run python -c "exit(3"

```

File: pipenv/cli.py (hunk 2 above)
Comment (nonylene) [inline pipenv/cli.py]: Oh, I have no reason!

fixed : 2fe5d2e

**Pull request merged**